starknet-rpc-test/get_transaction_receipt.rs

```diff
@@ -265,3 +268,91 @@ async fn fail_invalid_transaction_hash(#[future] madara: MadaraClient) -> Result
 
     Ok(())
 }
+
+#[rstest]
+#[tokio::test]
+async fn work_with_messages_to_l1(#[future] madara: MadaraClient) -> Result<(), anyhow::Error> {
+    let madara = madara.await;
+    let rpc = madara.get_starknet_client();
+
+    // 1. Declaring class for our L2 > L1 contract
+
+    let account = create_account(rpc, SIGNER_PRIVATE, ARGENT_CONTRACT_ADDRESS, true);
+    let (declare_tx, _) = account.declare_legacy_contract("../cairo-contracts/build/send_message.json");
+
+    let mut txs = madara.create_block_with_txs(vec![Transaction::LegacyDeclaration(declare_tx)]).await?;
+
+    // 2. Determine class hash
+
+    let class_hash = match txs.remove(0).unwrap() {
+        TransactionResult::Declaration(rpc_response) => rpc_response.class_hash,
+        _ => panic!("expected execution result"),
+    };
+
+    // 3. Next, deploying an instance of this class using universal deployer
+
+    let deploy_tx = account.call_contract(
```

Comment:
In starknet `calls` are not supposed to modify the state. 
Therefore it is not supposed to actually deploy the contract. Just to do a `dry` run

Reply:
Ah ok, it's just the matter of naming or other changes required?

Reply:
If you are doing an invoke and not a call, yes just rename it will work